### ISSUE TITLE
Step ID is not unique

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#209]
### Version

0.3.1

### Reproduction steps

1. File `.github/workflows/x.yml` with the following content:
   ```yaml
   on: push
   jobs:
     repro:
       runs-on: ubuntu-latest
       steps:
         - uses: actions/github-script@v7
           with:
             script: |
               const title = "${'$'}{{ github.event.pull_request.title }}";
               return title;
         - uses: actions/github-script@v7
           with:
             script: |
               const title = "${'$'}{{ github.event.pull_request.title }}";
               return title;
   ```

2. Run `ghlint` with the following arguments:
   ```shell
   ghlint --console .github/workflows/x.yml
   ```

### Expected behavior

Ability to distinguish which step is targeted.

### Actual behavior

```
Step[actions/github-script@v7] in Job[test] JavaScript contains GitHub Expressions.
Step[actions/github-script@v7] in Job[test] JavaScript contains GitHub Expressions.
```

### Additional Context

See `TODO re-think identifier, there could be multiple like this, it's not unique` in code.
